### ISSUE TITLE
How to work on standalone files (not in a swift package)?

### ISSUE THREAD

**Issue by lakca** [apple/sourcekit-lsp#728]
When I edit a single swift file inside vscode, I got the following error:<img width="570" alt="image" src="https://user-images.githubusercontent.com/16255922/231135077-3be7cbf4-1672-49b9-88f8-9705c774dd36.png">

**Comment by adam-fowler** [comment]
What version of Swift are you running? With 5.8 on macOS it seems to work fine and the last time I checked it was working on Ubuntu 2204.

**Comment by lakca** [comment]
@adam-fowler 
Apple Swift version 5.7.2 (swiftlang-5.7.2.135.5 clang-1400.0.29.51)
Target: arm64-apple-macosx13.0

**Comment by MaxDesiatov** [comment]
Not reproducible for me with Swift 5.8 from Xcode 14.3.0 and Swift extension v1.2.0 either. I think this can be closed as outdated.

**Comment by lakca** [comment]
Xcode is also okay for me. The error was reported from VSCode.

**Comment by MaxDesiatov** [comment]
> Xcode is also okay for me. The error was reported from VSCode.

I understand that, I only mentioned Xcode as a distribution mechanism for Swift 5.8. I did verify that the issue is not reproducible in VS Code with that specific Swift version. Can you confirm that this is the case with that version for you too?

**Comment by ahoppen** [comment]
If it still reproduces, could you share any sourcekit-lsp crash reports in the Console app?

**Comment by lakca** [comment]
> If it still reproduces, could you share any sourcekit-lsp crash reports in the Console app?

@ahoppen Here's the crash report in the console:：

```
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               sourcekit-lsp [75326]
Path:                  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/sourcekit-lsp
Identifier:            sourcekit-lsp
Version:               ???
Code Type:             ARM-64 (Native)
Parent Process:        Code Helper (Plugin) [75149]
Responsible:           Electron [25960]
User ID:               501

Date/Time:             2023-04-11 18:45:56.2013 +0800
OS Version:            macOS 13.2 (22D49)
Report Version:        12
Anonymous UUID:        6944BC15-890C-D809-E667-FD2105AB133D

Sleep/Wake UUID:       EB19E2D4-EFBD-4929-AE61-84C020F6D4A8

Time Awake Since Boot: 640000 seconds
Time Since Wake:       119866 seconds

System Integrity Protection: enabled

Crashed Thread:        1  Dispatch queue: swift-language-server-queue

Exception Type:        EXC_BREAKPOINT (SIGTRAP)
Exception Codes:       0x0000000000000001, 0x0000000101a7f14c

Termination Reason:    Namespace SIGNAL, Code 5 Trace/BPT trap: 5
Terminating Process:   exc handler [75326]

Thread 0:
0   libsystem_kernel.dylib        	       0x184f996ec __semwait_signal_nocancel + 8
1   libsystem_c.dylib             	       0x184ebbc4c nanosleep$NOCANCEL + 216
2   libsystem_c.dylib             	       0x184ebbb18 sleep$NOCANCEL + 52
3   libdispatch.dylib             	       0x184e5dbb0 _dispatch_queue_cleanup2 + 212
4   libsystem_pthread.dylib       	       0x184fcf9ec _pthread_tsd_cleanup + 620
5   libsystem_pthread.dylib       	       0x184fd2760 _pthread_exit + 84
6   libsystem_pthread.dylib       	       0x184fcf71c pthread_exit + 88
7   libdispatch.dylib             	       0x184e5908c dispatch_main + 128
8   sourcekit-lsp                 	       0x100e35cd0 0x100e30000 + 23760
9   sourcekit-lsp                 	       0x100e37e64 0x100e30000 + 32356
10  sourcekit-lsp                 	       0x101ad408c 0x100e30000 + 13254796
11  sourcekit-lsp                 	       0x100e33bf8 0x100e30000 + 15352
12  dyld                          	       0x184ca7e50 start + 2544

Thread 1 Crashed::  Dispatch queue: swift-language-server-queue
0   sourcekit-lsp                 	       0x101a7f14c 0x100e30000 + 12906828
1   sourcekit-lsp                 	       0x101a7c110 0x100e30000 + 12894480
2   sourcekit-lsp                 	       0x101a7c110 0x100e30000 + 12894480
3   sourcekit-lsp                 	       0x101a7d4e4 0x100e30000 + 12899556
4   sourcekit-lsp                 	       0x101a7e8f8 0x100e30000 + 12904696
5   sourcekit-lsp                 	       0x101a8bd10 0x100e30000 + 12958992
6   sourcekit-lsp                 	       0x1010317e4 0x100e30000 + 2103268
7   sourcekit-lsp                 	       0x101a8b860 0x100e30000 + 12957792
8   sourcekit-lsp                 	       0x101a8c278 0x100e30000 + 12960376
9   sourcekit-lsp                 	       0x101a9fdc8 0x100e30000 + 13041096
10  sourcekit-lsp                 	       0x101a4c43c 0x100e30000 + 12698684
11  libdispatch.dylib             	       0x184e489dc _dispatch_call_block_and_release + 32
12  libdispatch.dylib             	       0x184e4a504 _dispatch_client_callout + 20
13  libdispatch.dylib             	       0x184e51bbc _dispatch_lane_serial_drain + 672
14  libdispatch.dylib             	       0x184e52708 _dispatch_lane_invoke + 384
15  libdispatch.dylib             	       0x184e5d34c _dispatch_workloop_worker_thread + 652
16  libsystem_pthread.dylib       	       0x184fce100 _pthread_wqthread + 288
17  libsystem_pthread.dylib       	       0x184fcce20 start_wqthread + 8

Thread 2:
0   libsystem_pthread.dylib       	       0x184fcce18 start_wqthread + 0

Thread 3:
0   libsystem_pthread.dylib       	       0x184fcce18 start_wqthread + 0

Thread 4:
0   libsystem_kernel.dylib        	       0x184f99724 __sigsuspend_nocancel + 8
1   libdispatch.dylib             	       0x184e5dcb0 _dispatch_sigsuspend + 48
2   libdispatch.dylib             	       0x184e5dc80 _dispatch_sig_thread + 60

Thread 5:
0   libsystem_pthread.dylib       	       0x184fcce18 start_wqthread + 0


Thread 1 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000000   x1: 0xe000000000000000   x2: 0x0000000000000000   x3: 0xe000000000000000
    x4: 0x000000000000004f   x5: 0x0000000000000004   x6: 0x0000000000000002   x7: 0x00000001e15f2048
    x8: 0x0000000000000000   x9: 0x0000000000000000  x10: 0x0000000000000000  x11: 0x000000000000001f
   x12: 0x0000000000000016  x13: 0x000000011c712fc0  x14: 0x01000001e05745b1  x15: 0x00000001e05745b0
   x16: 0x000000019317c504  x17: 0xf948000185052114  x18: 0x0000000000000000  x19: 0xe000000000000000
   x20: 0x0000000000000000  x21: 0x0000000000000000  x22: 0xe000000000000000  x23: 0x000000016f053660
   x24: 0x0000000100e39e68  x25: 0x0000000101d715b8  x26: 0x0000000100f171a8  x27: 0x0000000100f171a0
   x28: 0x00000000000001a3   fp: 0x000000016f0535e0   lr: 0x0000000101a7c110
    sp: 0x000000016f0535a0   pc: 0x0000000101a7f14c cpsr: 0x00001000
   far: 0x00000001411c6f95  esr: 0xf2000001 (Breakpoint) brk 1

Binary Images:
       0x184f91000 -        0x184fcafeb libsystem_kernel.dylib (*) <6b682e21-fd96-3a5a-8260-fce367258252> /usr/lib/system/libsystem_kernel.dylib
       0x184e91000 -        0x184f11ffb libsystem_c.dylib (*) <14cd841b-0c7b-34a2-a342-cc6796ef9259> /usr/lib/system/libsystem_c.dylib
       0x184e46000 -        0x184e8dfff libdispatch.dylib (*) <4d52c3a5-32e6-37a2-9d6c-23c612038354> /usr/lib/system/libdispatch.dylib
       0x184fcb000 -        0x184fd7ffb libsystem_pthread.dylib (*) <9f3b729a-ed04-3e65-adac-d75ad06ebbdc> /usr/lib/system/libsystem_pthread.dylib
       0x100e30000 -        0x101d5ffff sourcekit-lsp (*) <7e13587e-8baf-36e4-af6d-06d67bc58c5f> /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/sourcekit-lsp
       0x184ca2000 -        0x184d2cba3 dyld (*) <fe8a9d9e-f65d-34ca-942c-175b99c0601b> /usr/lib/dyld

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 10
    thread_create: 1
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=943.3M resident=0K(0%) swapped_out_or_unallocated=943.3M(100%)
Writable regions: Total=1.1G written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=1.1G(100%)

                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Activity Tracing                   256K        1 
Dispatch continuations            64.0M        1 
Kernel Alloc Once                   32K        1 
MALLOC                           136.1M       30 
MALLOC guard page                   96K        5 
MALLOC_MEDIUM (reserved)         960.0M        8         reserved VM address space (unallocated)
STACK GUARD                       56.1M        6 
Stack                             10.6M        6 
VM_ALLOCATE                         16K        1 
__AUTH                             461K      102 
__AUTH_CONST                      8023K      230 
__DATA                            3400K      219 
__DATA_CONST                      9828K      232 
__DATA_DIRTY                       445K       84 
__FONT_DATA                        2352        1 
__LINKEDIT                       769.5M        3 
__OBJC_CONST                       643K       81 
__OBJC_RO                         65.5M        1 
__OBJC_RW                         1988K        1 
__TEXT                           173.8M      242 
dyld private memory                256K        1 
mapped file                       29.3M        3 
shared memory                      624K        7 
===========                     =======  ======= 
TOTAL                              2.2G     1266 
TOTAL, minus reserved VM space     1.3G     1266 



-----------
Full Report
-----------

{"app_name":"sourcekit-lsp","timestamp":"2023-04-11 18:45:57.00 +0800","app_version":"","slice_uuid":"7e13587e-8baf-36e4-af6d-06d67bc58c5f","build_version":"","platform":1,"share_with_app_devs":0,"is_first_party":1,"bug_type":"309","os_version":"macOS 13.2 (22D49)","roots_installed":0,"incident_id":"1B045DD5-6228-4D68-8189-0370C3160A92","name":"sourcekit-lsp"}
{
  "uptime" : 640000,
  "procRole" : "Unspecified",
  "version" : 2,
  "userID" : 501,
  "deployVersion" : 210,
  "modelCode" : "MacBookAir10,1",
  "coalitionID" : 2341,
  "osVersion" : {
    "train" : "macOS 13.2",
    "build" : "22D49",
    "releaseType" : "User"
  },
  "captureTime" : "2023-04-11 18:45:56.2013 +0800",
  "incident" : "1B045DD5-6228-4D68-8189-0370C3160A92",
  "pid" : 75326,
  "translated" : false,
  "cpuType" : "ARM-64",
  "roots_installed" : 0,
  "bug_type" : "309",
  "procLaunch" : "2023-04-11 18:45:55.9385 +0800",
  "procStartAbsTime" : 15396284276582,
  "procExitAbsTime" : 15396290545386,
  "procName" : "sourcekit-lsp",
  "procPath" : "\/Applications\/Xcode.app\/Contents\/Developer\/Toolchains\/XcodeDefault.xctoolchain\/usr\/bin\/sourcekit-lsp",
  "parentProc" : "Code Helper (Plugin)",
  "parentPid" : 75149,
  "coalitionName" : "com.microsoft.VSCode",
  "crashReporterKey" : "6944BC15-890C-D809-E667-FD2105AB133D",
  "responsiblePid" : 25960,
  "responsibleProc" : "Electron",
  "wakeTime" : 119866,
  "sleepWakeUUID" : "EB19E2D4-EFBD-4929-AE61-84C020F6D4A8",
  "sip" : "enabled",
  "exception" : {"codes":"0x0000000000000001, 0x0000000101a7f14c","rawCodes":[1,4322750796],"type":"EXC_BREAKPOINT","signal":"SIGTRAP"},
  "termination" : {"flags":0,"code":5,"namespace":"SIGNAL","indicator":"Trace\/BPT trap: 5","byProc":"exc handler","byPid":75326},
  "os_fault" : {"process":"sourcekit-lsp"},
  "extMods" : {"caller":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"system":{"thread_create":1,"thread_set_state":0,"task_for_pid":10},"targeted":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"warnings":0},
  "faultingThread" : 1,
  "threads" : [{"id":8280250,"frames":[{"imageOffset":34540,"symbol":"__semwait_signal_nocancel","symbolLocation":8,"imageIndex":0},{"imageOffset":175180,"symbol":"nanosleep$NOCANCEL","symbolLocation":216,"imageIndex":1},{"imageOffset":174872,"symbol":"sleep$NOCANCEL","symbolLocation":52,"imageIndex":1},{"imageOffset":97200,"symbol":"_dispatch_queue_cleanup2","symbolLocation":212,"imageIndex":2},{"imageOffset":18924,"symbol":"_pthread_tsd_cleanup","symbolLocation":620,"imageIndex":3},{"imageOffset":30560,"symbol":"_pthread_exit","symbolLocation":84,"imageIndex":3},{"imageOffset":18204,"symbol":"pthread_exit","symbolLocation":88,"imageIndex":3},{"imageOffset":77964,"symbol":"dispatch_main","symbolLocation":128,"imageIndex":2},{"imageOffset":23760,"imageIndex":4},{"imageOffset":32356,"imageIndex":4},{"imageOffset":13254796,"imageIndex":4},{"imageOffset":15352,"imageIndex":4},{"imageOffset":24144,"symbol":"start","symbolLocation":2544,"imageIndex":5}]},{"triggered":true,"id":8280251,"threadState":{"x":[{"value":0},{"value":16140901064495857664},{"value":0},{"value":16140901064495857664},{"value":79},{"value":4},{"value":2},{"value":8076075080,"symbolLocation":0,"symbol":"type metadata for String"},{"value":0},{"value":0},{"value":0},{"value":31},{"value":22},{"value":4772147136},{"value":72057602096711089,"symbolLocation":72057594037927937,"symbol":"OBJC_CLASS_$___NSCFString"},{"value":8058783152,"symbolLocation":0,"symbol":"OBJC_CLASS_$___NSCFString"},{"value":6762775812,"symbolLocation":0,"symbol":"swift_bridgeObjectRelease"},{"value":17962607120293896468,"symbolLocation":17962607113767223296,"symbol":"-[__NSCFString release]"},{"value":0},{"value":16140901064495857664},{"value":0},{"value":0},{"value":16140901064495857664},{"value":6157579872},{"value":4309884520},{"value":4325840312},{"value":4310790568},{"value":4310790560},{"value":419}],"flavor":"ARM_THREAD_STATE64","lr":{"value":4322738448},"cpsr":{"value":4096},"fp":{"value":6157579744},"sp":{"value":6157579680},"esr":{"value":4060086273,"description":"(Breakpoint) brk 1"},"pc":{"value":4322750796,"matchesCrashFrame":1},"far":{"value":5387349909}},"queue":"swift-language-server-queue","frames":[{"imageOffset":12906828,"imageIndex":4},{"imageOffset":12894480,"imageIndex":4},{"imageOffset":12894480,"imageIndex":4},{"imageOffset":12899556,"imageIndex":4},{"imageOffset":12904696,"imageIndex":4},{"imageOffset":12958992,"imageIndex":4},{"imageOffset":2103268,"imageIndex":4},{"imageOffset":12957792,"imageIndex":4},{"imageOffset":12960376,"imageIndex":4},{"imageOffset":13041096,"imageIndex":4},{"imageOffset":12698684,"imageIndex":4},{"imageOffset":10716,"symbol":"_dispatch_call_block_and_release","symbolLocation":32,"imageIndex":2},{"imageOffset":17668,"symbol":"_dispatch_client_callout","symbolLocation":20,"imageIndex":2},{"imageOffset":48060,"symbol":"_dispatch_lane_serial_drain","symbolLocation":672,"imageIndex":2},{"imageOffset":50952,"symbol":"_dispatch_lane_invoke","symbolLocation":384,"imageIndex":2},{"imageOffset":95052,"symbol":"_dispatch_workloop_worker_thread","symbolLocation":652,"imageIndex":2},{"imageOffset":12544,"symbol":"_pthread_wqthread","symbolLocation":288,"imageIndex":3},{"imageOffset":7712,"symbol":"start_wqthread","symbolLocation":8,"imageIndex":3}]},{"id":8280252,"frames":[{"imageOffset":7704,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":3}]},{"id":8280253,"frames":[{"imageOffset":7704,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":3}]},{"id":8280270,"frames":[{"imageOffset":34596,"symbol":"__sigsuspend_nocancel","symbolLocation":8,"imageIndex":0},{"imageOffset":97456,"symbol":"_dispatch_sigsuspend","symbolLocation":48,"imageIndex":2},{"imageOffset":97408,"symbol":"_dispatch_sig_thread","symbolLocation":60,"imageIndex":2}]},{"id":8280271,"frames":[{"imageOffset":7704,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":3}]}],
  "usedImages" : [
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6525882368,
    "size" : 237548,
    "uuid" : "6b682e21-fd96-3a5a-8260-fce367258252",
    "path" : "\/usr\/lib\/system\/libsystem_kernel.dylib",
    "name" : "libsystem_kernel.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6524833792,
    "size" : 528380,
    "uuid" : "14cd841b-0c7b-34a2-a342-cc6796ef9259",
    "path" : "\/usr\/lib\/system\/libsystem_c.dylib",
    "name" : "libsystem_c.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6524526592,
    "size" : 294912,
    "uuid" : "4d52c3a5-32e6-37a2-9d6c-23c612038354",
    "path" : "\/usr\/lib\/system\/libdispatch.dylib",
    "name" : "libdispatch.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6526119936,
    "size" : 53244,
    "uuid" : "9f3b729a-ed04-3e65-adac-d75ad06ebbdc",
    "path" : "\/usr\/lib\/system\/libsystem_pthread.dylib",
    "name" : "libsystem_pthread.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4309843968,
    "size" : 15925248,
    "uuid" : "7e13587e-8baf-36e4-af6d-06d67bc58c5f",
    "path" : "\/Applications\/Xcode.app\/Contents\/Developer\/Toolchains\/XcodeDefault.xctoolchain\/usr\/bin\/sourcekit-lsp",
    "name" : "sourcekit-lsp"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6522806272,
    "size" : 568228,
    "uuid" : "fe8a9d9e-f65d-34ca-942c-175b99c0601b",
    "path" : "\/usr\/lib\/dyld",
    "name" : "dyld"
  }
],
  "sharedCache" : {
  "base" : 6522159104,
  "size" : 3447406592,
  "uuid" : "3366b98c-6b8a-3546-8233-dc167320439f"
},
  "vmSummary" : "ReadOnly portion of Libraries: Total=943.3M resident=0K(0%) swapped_out_or_unallocated=943.3M(100%)\nWritable regions: Total=1.1G written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=1.1G(100%)\n\n                                VIRTUAL   REGION \nREGION TYPE                        SIZE    COUNT (non-coalesced) \n===========                     =======  ======= \nActivity Tracing                   256K        1 \nDispatch continuations            64.0M        1 \nKernel Alloc Once                   32K        1 \nMALLOC                           136.1M       30 \nMALLOC guard page                   96K        5 \nMALLOC_MEDIUM (reserved)         960.0M        8         reserved VM address space (unallocated)\nSTACK GUARD                       56.1M        6 \nStack                             10.6M        6 \nVM_ALLOCATE                         16K        1 \n__AUTH                             461K      102 \n__AUTH_CONST                      8023K      230 \n__DATA                            3400K      219 \n__DATA_CONST                      9828K      232 \n__DATA_DIRTY                       445K       84 \n__FONT_DATA                        2352        1 \n__LINKEDIT                       769.5M        3 \n__OBJC_CONST                       643K       81 \n__OBJC_RO                         65.5M        1 \n__OBJC_RW                         1988K        1 \n__TEXT                           173.8M      242 \ndyld private memory                256K        1 \nmapped file                       29.3M        3 \nshared memory                      624K        7 \n===========                     =======  ======= \nTOTAL                              2.2G     1266 \nTOTAL, minus reserved VM space     1.3G     1266 \n",
  "legacyInfo" : {
  "threadTriggered" : {
    "queue" : "swift-language-server-queue"
  }
},
  "trialInfo" : {
  "rollouts" : [
    {
      "rolloutId" : "5f72dc58705eff005a46b3a9",
      "factorPackIds" : {

      },
      "deploymentId" : 240000015
    },
    {
      "rolloutId" : "6112d17137f5d11121dcd4e2",
      "factorPackIds" : {

      },
      "deploymentId" : 240000250
    }
  ],
  "experiments" : [
    {
      "treatmentId" : "c28e4ee6-1b08-4f90-8e05-2809e78310a3",
      "experimentId" : "6317d2003d24842ff850182a",
      "deploymentId" : 400000013
    }
  ]
}
}
```

<img width="473" alt="image" src="https://user-images.githubusercontent.com/16255922/231324800-02a2b125-1fbd-4539-b868-bef3b2567cd2.png">

****

@MaxDesiatov  I don't know what happened, but indeed when I reopened the project in VSCode this morning, there was no crush anymore.

**Comment by ahoppen** [comment]
I see what’s going wrong. You are hitting https://github.com/apple/sourcekit-lsp/issues/635. This should be fixed with Swift 5.8, which is included in Xcode 14.3. If you are still able to reproduce the problem with that version of SourceKit-LSP, please re-open the issue.
